Homework/app/models/cleaner.rb

```diff
@@ -1,10 +1,14 @@
 class Cleaner < ApplicationRecord
-  has_and_belongs_to_many :cities
+  has_and_belongs_to_many :cities,dependent: :destroy
+  has_many :bookings,dependent: :destroy
 
   validates :first_name,presence:true
   validates :last_name,presence:true
   validates :quality_score,presence:true
   validates :email,presence:true
+  validates :email, format: { with: /\A[\w+\-.]+@[a-z\d\-]+(\.[a-z\d\-]+)*\.[a-z]+\z/,
```

Comment:
Use constants like EMAIL_REGEX = /\A[\w+\-.]+@[a-z\d\-]+(\.[a-z\d\-]+)*\.[a-z]+\z/

For all those values which are constants.